operations/security_and_authentication/Audit_Logs.md

```diff
@@ -0,0 +1,65 @@
+# Audit Logs
+
+How to enable and disable audit logging for host and Kubernetes APIs.
+
+Audit logs are used to monitor the system and search for suspicious behavior.
+Host and Kubernetes API audit logging can be enabled to produce extra audit logs for analysis.
+Enabling audit logging is optional and will generate some load and data on the non-compute nodes \(NCNs\).
+Host and Kubernetes API audit logging is not enabled by default.
+
+## Procedure
+
+1. Enable or disable host and Kubernetes API audit logging.
+
+   The method for updating the audit log settings will vary depending on the state of the system.
+   Select one of the following options based on the installation status of the system:
+
+   * Install
+
+     During the install, audit logging is enabled or disabled by changing the CSI settings.
+     Use the following flags with the `csi config init` command to enable or disable audit logging.
+     Refer to `csi config init -h` for more information on using flags.
+
+     * `ncn-mgmt-node-auditing-enabled`: Set to `true` to enable host logging or to `false` to disable host logging.
+
+       ```console
+       initCmd.Flags().Bool("ncn-mgmt-node-auditing-enabled", true, "Enable management node auditing")
+       ```
+
+     * `k8s-api-auditing-enabled`: Set to `true` to enable Kubernetes API logging or to `false` to disable Kubernetes API logging.
+
+       ```console
+       initCmd.Flags().Bool("k8s-api-auditing-enabled", true, "Enable the kubernetes auditing API")
+       ```
+
+   * Post-install
+
+     If a system is installed, audit logging can be enabled via the Boot Script Service (BSS).
+
+     * `ncn-mgmt-node-auditing-enabled`
+     * `k8s-api-auditing-enabled`
+
+1. (`ncn#`) Verify if audit logging is enabled.
+
+   * `ncn-mgmt-node-auditing-enabled`
+  
+     ```bash
+     cray bss bootparameters list --format json --hosts Global | jq '.[]."cloud-init"."meta-data"."k8s-api-auditing-enabled"'
+     ```
+
+   * `k8s-api-auditing-enabled`
+
+     ```bash
+     cray bss bootparameters list --format json --hosts Global | jq '.[]."cloud-init"."meta-data"."ncn-mgmt-node-auditing-enabled"'
+     ```
```

Comment:
These will not be enabled by default, so this section should explain how to enable them.

Unfortunately, I'm not sure we have a way to update this global meta-data.

There is this, in `prerequisites.sh`, but it's not quite what you want as it is not a generic method to update meta-data.  It is very specific to the 1.0.x to csm-1.2.x upgrade path.

```
    # create the updated json
    jq '."cloud-init"."meta-data".host_records['$entry_number']|= . + {"aliases": ["packages.local", "registry.local"],"ip": "'$ip'"}' cloud-init-global.json  > cloud-init-global_update.json

    # post the update json to bss
    curl -s -k -H "Authorization: Bearer ${TOKEN}" --header "Content-Type: application/json" \
        --request PUT \
        --data @cloud-init-global_update.json \
        https://api-gw-service-nmn.local/apis/bss/boot/v1/bootparameters

    csi upgrade metadata --1-0-to-1-2 \
        --k8s-version ${KUBERNETES_VERSION} \
        --storage-version ${CEPH_VERSION}
```

In theory, there should be a way to update the meta-data, but it would probably require using curl to PATCH the datastructure.

@dzou-hpe - any ideas here?

Reply:
yes, i think we would need some curl to path that for upgrade

Reply:
@dzou-hpe / @heemstra any idea of what a more general curl command would be to update the metadata?